ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rewrite/AlterMaterializedViewRewriteAnalyzer.java

```diff
@@ -68,10 +68,12 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     Table materializedViewTable = getTable(tableName, true);
 
     // One last test: if we are enabling the rewrite, we need to check that query
-    // only uses transactional (MM and ACID) tables
+    // only uses transactional (MM and ACID and Iceberg) tables
     if (rewriteEnable) {
       for (SourceTable sourceTable : materializedViewTable.getMVMetadata().getSourceTables()) {
-        if (!AcidUtils.isTransactionalTable(sourceTable.getTable())) {
+        Table table = new Table(sourceTable.getTable());
+        if (!AcidUtils.isTransactionalTable(sourceTable.getTable()) &&
+                !(table.isNonNative() && table.getStorageHandler().areSnapshotsSupported())) {
           throw new SemanticException("Automatic rewriting for materialized view cannot be enabled if the " +
               "materialized view uses non-transactional tables");
         }
```

Comment:
Do we consider an Iceberg table as transactional? Should we change the message to something more general. For example:
`Cannot enable materialized view rewriting cause view uses unsupported tables`

Reply:
Yes, Iceberg tables are transactional but transactions are handled in a different than native transactional tables.